core/src/player.rs

```diff
@@ -1252,15 +1252,59 @@ impl Player {
                 ));
             }
 
+            let mut new_over_object_updated = false;
             // Cancel hover if an object is removed from the stage.
             if let Some(hovered) = context.mouse_over_object {
                 if !context.is_action_script_3() && hovered.as_displayobject().avm1_removed() {
                     context.mouse_over_object = None;
+                    if let Some(new_object) = new_over_object {
+                        if new_object.as_displayobject().id() == hovered.as_displayobject().id()
+                            && new_object.as_displayobject().depth()
+                                == hovered.as_displayobject().depth()
+                            && Arc::ptr_eq(
+                                &new_object.as_displayobject().movie(),
+                                &hovered.as_displayobject().movie(),
+                            )
+                        {
+                            if let Some(state) = hovered.as_displayobject().state() {
+                                new_object.as_displayobject().set_state(context, state);
+                            }
+                            context.mouse_over_object = Some(new_object);
+                            new_over_object_updated = true;
+                        }
+                    }
                 }
             }
-            if let Some(pressed) = context.mouse_down_object {
-                if !context.is_action_script_3() && pressed.as_displayobject().avm1_removed() {
+            if let Some(down_object) = context.mouse_down_object {
+                if !context.is_action_script_3() && down_object.as_displayobject().avm1_removed() {
                     context.mouse_down_object = None;
+                    let mut display_object = None;
+
+                    if let Some(root_clip) = context.stage.root_clip() {
+                        display_object = Self::find_first_character_instance(
+                            root_clip,
+                            down_object.as_displayobject(),
+                        );
+                    }
+
+                    if let Some(new_object) = display_object {
+                        if new_object.id() == down_object.as_displayobject().id()
+                            && new_object.depth() == new_object.depth()
```

Comment:
looks like a copy-paste error

Reply:
I don't see why it looks like a copy paste error, changed the name new_object to new_down_object anyways.